drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -239,79 +231,47 @@ static void event_handler(nrfx_twim_evt_t const *p_event, void *p_context)
 	k_sem_give(&dev_data->completion_sync);
 }
 
-static int init_twim(const struct device *dev)
-{
-	const struct i2c_nrfx_twim_config *dev_config = dev->config;
-	struct i2c_nrfx_twim_data *dev_data = dev->data;
-	nrfx_err_t result = nrfx_twim_init(&dev_config->twim,
-					   &dev_data->twim_config,
-					   event_handler, dev_data);
-	if (result != NRFX_SUCCESS) {
-		LOG_ERR("Failed to initialize device: %s", dev->name);
-		return -EIO;
-	}
-
-	dev_data->twim_initialized = true;
-	return 0;
-}
-
-static void deinit_twim(const struct device *dev)
-{
-	const struct i2c_nrfx_twim_config *dev_config = dev->config;
-	struct i2c_nrfx_twim_data *dev_data = dev->data;
-
-	if (dev_data->twim_initialized) {
-		nrfx_twim_uninit(&dev_config->twim);
-		dev_data->twim_initialized = false;
-	}
-}
-
 static int i2c_nrfx_twim_configure(const struct device *dev,
 				   uint32_t i2c_config)
 {
-	struct i2c_nrfx_twim_data *dev_data = dev->data;
-	nrf_twim_frequency_t frequency;
+	const struct i2c_nrfx_twim_config *dev_config = dev->config;
 
 	if (I2C_ADDR_10_BITS & i2c_config) {
 		return -EINVAL;
 	}
 
 	switch (I2C_SPEED_GET(i2c_config)) {
 	case I2C_SPEED_STANDARD:
-		frequency = NRF_TWIM_FREQ_100K;
+		nrf_twim_frequency_set(dev_config->twim.p_twim,
+				       NRF_TWIM_FREQ_100K);
 		break;
 	case I2C_SPEED_FAST:
-		frequency = NRF_TWIM_FREQ_400K;
+		nrf_twim_frequency_set(dev_config->twim.p_twim,
+				       NRF_TWIM_FREQ_400K);
 		break;
 #if NRF_TWIM_HAS_1000_KHZ_FREQ
 	case I2C_SPEED_FAST_PLUS:
-		frequency = NRF_TWIM_FREQ_1000K;
+		nrf_twim_frequency_set(dev_config->twim.p_twim,
+				       NRF_TWIM_FREQ_1000K);
 		break;
 #endif
 	default:
 		LOG_ERR("unsupported speed");
 		return -EINVAL;
 	}
 
-	if (frequency != dev_data->twim_config.frequency) {
-		dev_data->twim_config.frequency = frequency;
```

Comment:
Since `twim_config` is not changed now, it should be moved to the `i2c_nrfx_twim_config` structure.

Reply:
good point